src/moonlink/src/storage/iceberg/rest_catalog_test.rs

```diff
@@ -103,3 +110,100 @@ async fn test_load_table() {
     let result_table_ident = result.identifier().clone();
     assert_eq!(table_ident, result_table_ident);
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_update_table_with_requirement_check_failed() {
+    let namespace = get_random_string();
+    let table = get_random_string();
+    let guard = RestCatalogTestGuard::new(namespace.clone(), Some(table.clone()))
+        .await
+        .unwrap_or_else(|_| panic!("Rest catalog test guard creation fail, namespace={namespace}"));
+    let rest_catalog_config = default_rest_catalog_config();
```

Comment:
`default_rest_catalog_config` might not be a good function name, since we have random catalog inside, maybe consider renaming to sth else?